core/views.py

```diff
@@ -14,11 +17,37 @@
 unsupported_method_err = {"message" : "request method not supported by url"}
 
 
+
+def date_fields(model):
+    '''
+    Return a list of strings that correspond to a model's
+    fields that are instances of DateTimeField
+    '''
+    def is_date_field(name):
+        return type(getattr(model, name).field) == DateTimeField
+    
+    field_names = [f.name for f in model._meta.fields]
+    return list(filter(is_date_field, field_names))
+
+
+
 def read_many(model):
     def request_handler(request):
         params = { key: request.GET.get(key) for key in request.GET }
         filter_set=None
 
+        # Parsing incoming date fields
+        for key in params:
+            for field in date_fields(model):
+                if key.startswith(field):
+                    try:
+                        naive_date = dateutil.parser.parse(params[key])
+                        params[key] = make_aware(naive_date)
+                    except:
+                        message = f"Invalid date string provided for field {key}"
+                        invalid_date_err = { "message": message }
+                        return JsonResponse({"errors": [invalid_date_err]})
```

Comment:
does invalid_date_error just return {"message" : message} ?

it doesn't seem like this code would generate the error message in your photo-- (I trust your photo but would like an explanation of what's going on with this before approving the change)

Reply:
The 2nd `message` (without double quotes) in `{ "message": message }` is a variable, and would be replaced with the value it was assigned on the previous line :)

Reply:
<img width="345" alt="Screen Shot 2020-08-24 at 1 53 18 PM" src="https://user-images.githubusercontent.com/31521775/91084180-32d6ec80-e611-11ea-8b78-47fae57f1a90.png">

Reply:
oh
duh
thx